File: frontend/tcc/src/app/models/dto/usuario-perfil-dto.ts
import { ProposicaoResumoDTO } from './proposicao-resumo-dto';

export interface UsuarioPerfilDTO {
  id: number;
  cpf: string;
  nome: string;
  email: string;
  cep: string | null;
  escolaridade: string | null;
  profissao: string | null;
  genero: string | null;
  dataNascimento: string | null;
  totalVereadoresSeguidos: number;
  proposicoesFavoritas: ProposicaoResumoDTO[];
}

File: backend/ms-java/src/main/java/com/clientes/clientes_TCC/domain/Usuario/UsuarioPerfilDTO.java
package com.clientes.clientes_TCC.domain.Usuario;

import com.clientes.clientes_TCC.domain.Proposicao.ProposicaoResumoDTO;

import java.time.LocalDate;
import java.util.List;

public record UsuarioPerfilDTO(
        Integer id,
        String cpf,
        String nome,
        String email,
        String cep,
        String escolaridade,
        String profissao,
        String genero,
        LocalDate dataNascimento,
        Integer totalVereadoresSeguidos,
        List<ProposicaoResumoDTO> proposicoesFavoritas
) {
}
